packages/yoroi-extension/app/api/ada/lib/storage/bridge/updateTransactions.js

```diff
@@ -371,10 +371,13 @@ export async function rawGetTransactions(
                     ?.get(rewardAddr.payment_cred());
                   // happens if the MIR is to another pot
                   if (rewardAmount == null) continue;
+                  const rewardAmountStr = rewardAmount.is_positive()
+                    ? rewardAmount.as_positive().to_str()
+                    : ('-' + rewardAmount.as_negative().to_str());
                   implicitOutputSum.add({
                     identifier: defaultToken.defaultIdentifier,
                     networkId: defaultToken.defaultNetworkId,
-                    amount: new BigNumber(String(rewardAmount.as_i32())),
+                    amount: new BigNumber(rewardAmountStr),
```

Comment:
The problem that is being solved here is that Rust type `Int` from the serialization-lib wraps an `i128` and so can be a positive or a negative big number. Previously it was calling `as_i32()` which might return an `undefined` in case the i128 value is out of the i32 bounds, the error is silently ignorred (which was not a great decision).

The Rust type `BigInt` from the serialisation-lib wraps a `u64` value so can only hold positive integer numbers. This is why `Int` implements these weird methods for `.is_positive` , `.as_positive` , and `.as_negative` . We have to work around this conversion from a signed to an unsigned type but it is much more secure about the value size boundaries.

For the next serlib version there will be a new method `Int::to_str` which will returned the string with a positive or negative value directly.